createdisk.sh

```diff
@@ -187,24 +164,26 @@ copy_additional_files "$1" "$libvirtDestDir"
 
 tar cSf - --sort=name "$libvirtDestDir" | xz --threads=0 >"$libvirtDestDir.$crcBundleSuffix"
 
-# HyperKit image generation
-# This must be done after the generation of libvirt image as it reuse some of
-# the content of $libvirtDestDir
-hyperkitDestDir="crc_hyperkit_${destDirSuffix}"
-mkdir "$hyperkitDestDir"
-generate_hyperkit_directory "$libvirtDestDir" "$hyperkitDestDir" "$1" "$kernel_release" "$kernel_cmd_line"
-
-tar cSf - --sort=name "$hyperkitDestDir" | xz --threads=0 >"$hyperkitDestDir.$crcBundleSuffix"
-
-# HyperV image generation
-#
-# This must be done after the generation of libvirt image as it reuses some of
-# the content of $libvirtDestDir
-hypervDestDir="crc_hyperv_${destDirSuffix}"
-mkdir "$hypervDestDir"
-generate_hyperv_directory "$libvirtDestDir" "$hypervDestDir"
-
-tar cSf - --sort=name "$hypervDestDir" | xz --threads=0 >"$hypervDestDir.$crcBundleSuffix"
+if [ "${ARCH}" != "ppc64le" ]; then
+        # HyperKit image generation
+        # This must be done after the generation of libvirt image as it reuse some of
+        # the content of $libvirtDestDir
+        hyperkitDestDir="crc_hyperkit_${destDirSuffix}"
+        mkdir "$hyperkitDestDir"
+        generate_hyperkit_directory "$libvirtDestDir" "$hyperkitDestDir" "$1" "$kernel_release" "$kernel_cmd_line"
+
+        tar cSf - --sort=name "$hyperkitDestDir" | xz --threads=0 >"$hyperkitDestDir.$crcBundleSuffix"
+
+        # HyperV image generation
+        #
+        # This must be done after the generation of libvirt image as it reuses some of
+        # the content of $libvirtDestDir
+        hypervDestDir="crc_hyperv_${destDirSuffix}"
+        mkdir "$hypervDestDir"
+        generate_hyperv_directory "$libvirtDestDir" "$hypervDestDir"
+
+        tar cSf - --sort=name "$hypervDestDir" | xz --threads=0 >"$hypervDestDir.$crcBundleSuffix"
+fi
 
```

Comment:
Might read better as
```
# HyperKit image generation
if [ "${ARCH}" != "ppc64le" ]; then
        # This must be done after the generation of libvirt image as it reuse some of
        # the content of $libvirtDestDir
        hyperkitDestDir="crc_hyperkit_${destDirSuffix}"
        mkdir "$hyperkitDestDir"
        generate_hyperkit_directory "$libvirtDestDir" "$hyperkitDestDir" "$1" "$kernel_release" "$kernel_cmd_line"

        tar cSf - --sort=name "$hyperkitDestDir" | xz --threads=0 >"$hyperkitDestDir.$crcBundleSuffix"
fi

# HyperV image generation
if [ "${ARCH}" != "ppc64le" ]; then
        # This must be done after the generation of libvirt image as it reuses some of
        # the content of $libvirtDestDir
        hypervDestDir="crc_hyperv_${destDirSuffix}"
        mkdir "$hypervDestDir"
        generate_hyperv_directory "$libvirtDestDir" "$hypervDestDir"

        tar cSf - --sort=name "$hypervDestDir" | xz --threads=0 >"$hypervDestDir.$crcBundleSuffix"
fi
```

Once there, I'd even rename `generate_hyper*_directory` to `generate_hyper*_bundle` and move the ppc64le test + mkdir + tar calls there. But this bit can be done later.

Reply:
thanks - i like the idea of separate funcs for the different bundles but for now, ill split this ARCH check like in your comment. i agree its easier to read